sda-commons-server-swagger-example/src/main/java/org/sdase/commons/server/swagger/example/people/rest/PersonEndpoint.java

```diff
@@ -88,4 +90,10 @@ private URI getPersonUri(int id) {
         .resolveTemplate("personId", id)
         .build();
   }
+
+  @Override
+  public boolean configure(FeatureContext context) {
+    // TODO: Add comment here
+    return true;
+  }
```

Comment:
Maybe we can add an interface like `SdaEndpoint` or `ContextAware` or ??? for convenience that extends `Feature` and implements `configure` as default method?

Reply:
How about `ContextAwareEndpoint`? Though this would also be usable by any other class that is no JAX-RS resource.

Reply:
There is one thing that came to my mind this morning. Does it matter if the endpoint is context aware? From my understanding, Dropwizard 2 will reject all instances that do not implement _Feature_ (or similar) at the time of registration? So maybe _Endpoint_ is enough.

Although any interface for this purpose can be used by any service, I would recommend it for endpoints only. In some cases it may be useful for services that are only used by endpoints for the purpose of handling context information. But they may also be implemented with methods that convert a given context specific object into the desired value. If we keep all service context unaware and give the required information when calling a method, testing of the services should be much easier and we only have to take care about the context (mock it or build integration tests) when testing endpoints.

Reply:
It's not that Dropwizard 2 _rejects_ instances. It's just how they register it to jersey.

Some Context: When calling `environment.jersey().register(new Test())`, and `Test` is not identified as any kind of "Provider" by jersey, they create a new `AbstractBinder` that does `bind(object).to(clazz)`. But jersey will never inject any fields or contexts into `object`. A trivial solution would be to call `injectionManager.inject(object)` manually in the Dropwizard code, however, there is no trivial way to get access to Jersey's `InjectionManager` in an `AbstractBinder`.

Reply:
Thanks for the details. In this case `ContextAwareEndpoint` sounds most valuable as you suggested. It clearly names the purpose of the interface.